### DISCUSSION TITLE
"non-conforming drawing primitive definition" error when converting

### DISCUSSION THREAD

**Discussion by divilian** [ImageMagick/ImageMagick#2033] [Help] [answered]
I'm running ImageMagick 6.9.7-4 Q16 x86_64's `convert` utility on this `.svg` file:
```
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg"
     xmlns:xlink="http://www.w3.org/1999/xlink"
     version="1.2"
     width="40mm" height="40mm" viewBox="0 0 40 40"
     stroke="none"
     fill="#000000"
     stroke-width="0.3"
     font-size="3.88"
>
<defs>
  <marker id="arrow" markerWidth="15" markerHeight="7" refX="5" refY="3.5" orient="auto" markerUnits="strokeWidth">
    <path d="M0,0 L15,3.5 L0,7 z" stroke="context-stroke" fill="context-stroke"/>
  </marker>
</defs>
<g fill="#FF6347" id="img-8e49394f-1">
  <g transform="translate(20,20)">
    <path d="M-20,-20 L 20 -20 20 20 -20 20 z" class="primitive"/>
  </g>
</g>
</svg>
```
and I get these errors:
```
$ convert tomato.svg tomato.gif
convert-im6.q16: unrecognized color `context-stroke' @ warning/color.c/GetColorCompliance/1052.
convert-im6.q16: unable to open image `context-stroke': No such file or directory @ error/blob.c/OpenBlob/2701.
convert-im6.q16: no decode delegate for this image format `' @ error/constitute.c/ReadImage/504.
convert-im6.q16: non-conforming drawing primitive definition `stroke' @ error/draw.c/DrawImage/3265.
```
Any obvious ideas? I could create a bug in the Issues system, but wanted to check my sanity here first.
By the way, this is on Ubuntu 18.04.4 LTS.

[tomato.svg.txt](https://github.com/ImageMagick/ImageMagick/files/4647894/tomato.svg.txt)

**Comment by fmw42** [comment, marked as answer]
You are likely using the internal ImageMagick MSVG renderer, which is not as advanced as RSVG delegate or Inkscape. I suggest you install Inkscape on your systems and then try again. If it fails add -verbose to your command.

convert -verbose tomato.svg tomato.gif

Then report back what you get as messages.

With Inkscape installed, your command works for me and I get:

![test](https://user-images.githubusercontent.com/13458218/82280406-ac40ae00-9943-11ea-9ed7-1e635e45a351.gif)

**Comment by divilian** [comment]
Installed inkscape and it worked perfectly! Thanks, @fmw42!
